### PR TITLE
SPACE()

### LINKED ISSUE #3040: SPACE() | Return a string of the specified number of spaces

### PR DESCRIPTION
**Pull request by mshauneu**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Impl for SPACE() func

## Changelog

- New Feature

## Related Issues

Fixes #3040 

## Test Plan

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/strings/space.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct SpaceFunction {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/space.rs:26]: We can use `String2NumberFunction` instead.

File: common/functions/src/scalars/strings/space.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+impl fmt::Display for SpaceFunction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+#[inline]
+fn space(count: &u64) -> Vec<u8> {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/space.rs:98]: better to reuse the buffer.

Comment (mshauneu) [reply common/functions/src/scalars/strings/space.rs:98]: sorry didn't catch idea 
every time when new value come from const or series[i] I need a spaces with diff length...

Comment (mshauneu) [reply common/functions/src/scalars/strings/space.rs:98]: can you explain more?

Comment (sundy-li) [reply common/functions/src/scalars/strings/space.rs:98]: Refer to `Base64EncodeFunction` in the codes.

Comment (mshauneu) [reply common/functions/src/scalars/strings/space.rs]: changed

**Review by sundy-li — APPROVED** [review]

**Pull request merged**